src/ripple/app/main/Main.cpp

```diff
@@ -213,12 +213,40 @@ class multi_selector
                 return true;
         return false;
     }
+
+    std::size_t
+    size() const
+    {
+        return selectors_.size();
+    }
 };
 
 namespace test {
 extern std::atomic<bool> envUseIPv4;
 }
 
+template <class Runner>
+static bool
+anyMissing(Runner& runner, multi_selector const& pred)
+{
+    if (runner.tests() == 0)
+    {
+        runner.add_failures(1);
+        std::cout << "Failed: No tests run" << std::endl;
+        return true;
+    }
+    if (runner.suites() < pred.size())
+    {
+        auto const missing = pred.size() - runner.suites();
+        runner.add_failures(missing);
+        std::cout << "Failed: " << missing
+                  << " filters did not match any existing test suites"
+                  << std::endl;
```

Comment:
Would it be worth mentioning which filters?

Reply:
I think it would be nice to mention which filters, but I was trying to keep these changes relatively simple, and my initial investigation into ways to either get the list of run tests or remove the run tests from the predicate was not simple, so I punted. I think that if it becomes enough of an issue, we can find a way to add it, but this seems sufficient for now. For now, it's easy enough to take the list of tests from the command line and rewrite it as a shell `for` loop to find the test that doesn't run anything.